ports/stm/peripherals/stm32f7/stm32f746xx/pins.c

```diff
@@ -28,12 +28,14 @@
 #include "py/mphal.h"
 #include "peripherals/pins.h"
 
-// Todo: some pins do have ADCs, but the module isn't set up yet.
+// Todo: Add ADC on all pins, ADC only added on A0-A5 for STM32F746ZG nucleo
```

Comment:
There are some python scripts in `stm/tools` that you could try tweaking to help simplify adding these from tables copied out of the datasheet or Micropython's CSV files.